compiler/rustc_resolve/src/imports.rs

```diff
@@ -922,11 +922,35 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
                         .note(&format!("consider declaring type or module `{}` with `pub`", ident))
                         .emit();
                 } else {
-                    let note_msg =
-                        format!("consider marking `{}` as `pub` in the imported module", ident);
-                    struct_span_err!(self.r.session, import.span, E0364, "{}", error_msg)
-                        .span_note(import.span, &note_msg)
-                        .emit();
+                    let mut err =
+                        struct_span_err!(self.r.session, import.span, E0364, "{error_msg}");
+                    match binding.kind {
+                        NameBindingKind::Res(Res::Def(DefKind::Macro(_), _def_id), _)
+                            // exclude decl_macro
+                            if !self.r.session.features_untracked().decl_macro
+                                || !self
+                                    .r
+                                    .session
+                                    .source_map()
+                                    .span_to_snippet(binding.span)
+                                    .map(|snippet| snippet.starts_with("macro "))
+                                    .unwrap_or(true) =>
```

Comment:
We try to avoid this kind of snippet based reasoning wherever we can. Is there another way?